packages/shared/src/hooks/utils/useQueryState.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  QueryKey,
+  useQuery,
+  useQueryClient,
+  UseQueryResult,
+} from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { disabledRefetch } from '../../lib/func';
+
+type UseQueryState<T> = [
+  T,
+  (value: T) => void,
+  Omit<UseQueryResult<T>, 'data'>,
+];
+
+export const useQueryState = <T>(
+  key: QueryKey,
+  defaultValue: T,
+): UseQueryState<T> => {
```

Comment:
Let's maybe make it object based (just in case we add more later)

Reply:
Yeah, this has no restriction, you can pass an object, boolean, string, or anything. Also we did not set a specific type.

Reply:
I mean instead of plain props make it object props :) 

```
export const useQueryState = <T>({
  key
  defaultValue
}: PropsDef): UseQueryState<T> => {
```

Reply:
Ahh, cool. You meant the props. I just thought of making it very similar to `useState`, but yeah, we can do that too.

Reply:
I was just thinking we can have the optional parameters on the third when we need it. But yeah, updated the props now.